src/triage/component/architect/feature_dictionary.py

```diff
@@ -0,0 +1,35 @@
+from triage.component.architect.utils import remove_schema_from_table_name
+from triage.util.structs import FeatureNameList
+from collections.abc import Iterable
+
+
+class FeatureDictionary(dict):
+    """A feature dictionary, consisting of table names as keys and column names as values
+
+    If a list of feature_blocks is passed, will initialize the feature dictionary with their data.
+    """
+    def __init__(self, feature_blocks=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        for feature_block in feature_blocks:
+            cleaned_table = remove_schema_from_table_name(
+                feature_block.final_feature_table_name
+            )
+            self[cleaned_table] = feature_block.feature_columns
+
+    def __setitem__(self, table, feature_names):
+        if not isinstance(table, str):
+            raise ValueError("key of FeatureDictionary objects represents a table "
+                             "name and must be a string")
+        if not isinstance(feature_names, Iterable):
+            raise ValueError("value of FeatureDictionary objects represents a list of "
+                             "feature names, and therefore must be iterable")
+        for feature_name in feature_names:
+            if not isinstance(feature_name, str):
```

Comment:
if you like, this check might make (more) sense in `FeatureNameList`

Reply:
Makes sense. I'm testing this out, and it's a bit of work to actually do right. The same concerns about inheriting from dict apply to inheriting from list, at least once you do what we're doing now and start to change the mutation methods to add something like this in.

https://stackoverflow.com/questions/3487434/overriding-append-method-after-inheriting-from-a-python-list

The top-rated answer makes a good suggestion; while inheriting from list, I've so far overridden `insert`, `__setitem__`, and `append` and still haven't found the one that hits how we actually use FeatureNameList. I assume it's one of `extend`, `__add__`, or `__iadd__`, but that's quite a bit to do to properly inherit from `list`. So implementing the suggestion in there for a typed list looks promising.

Of course, there's also the comment further below that raises the point that inheriting from MutableSequence also means that we have to figure out serialization on our own. This might not be a bad thing to do anyway: as it stands, this class doesn't serialize right and the calling code is forced to listify it. It would be nice to just implement this right in `FeatureNameList`.

Reply:
It ended up being that both `__add__` and `extend` are used by triage

Reply:
Makes sense. Again, doesn't abstractly _have_ to be one way or the other, (nor encapsulated in one place or another). But, good to consider :+1: